kie-drools-wb/kie-drools-wb-webapp/src/main/java/org/kie/workbench/drools/client/docks/AuthoringWorkbenchDocks.java

```diff
@@ -0,0 +1,67 @@
+import org.uberfire.client.workbench.docks.UberfireDock;
+import org.uberfire.client.workbench.docks.UberfireDockPosition;
+import org.uberfire.client.workbench.docks.UberfireDockReadyEvent;
+import org.uberfire.client.workbench.docks.UberfireDocks;
+import org.uberfire.mvp.PlaceRequest;
+import org.uberfire.mvp.impl.DefaultPlaceRequest;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+@Dependent
+public class AuthoringWorkbenchDocks {
+
+    @Inject
+    private UberfireDocks uberfireDocks;
+
+    @Inject
+    protected DataModelerWorkbenchContext dataModelerWBContext;
+
+    private String authoringPerspectiveIdentifier;
+
+    private UberfireDock projectExplorerDock;
+
+
+    public void perspectiveChangeEvent(@Observes UberfireDockReadyEvent dockReadyEvent) {
+        if (authoringPerspectiveIdentifier != null && dockReadyEvent.getCurrentPerspective().equals(authoringPerspectiveIdentifier)) {
+            if (projectExplorerDock != null) {
+                uberfireDocks.expand(projectExplorerDock);
+            }
+        }
+    }
+
+    public void setup(String authoringPerspectiveIdentifier, PlaceRequest projectExplorerPlaceRequest) {
+        this.authoringPerspectiveIdentifier = authoringPerspectiveIdentifier;
+        projectExplorerDock = new UberfireDock(UberfireDockPosition.WEST, "ADJUST", projectExplorerPlaceRequest, authoringPerspectiveIdentifier).withSize(400).withLabel("Project Explorer");
+        uberfireDocks.add(
+                projectExplorerDock,
+                new UberfireDock(UberfireDockPosition.EAST, "RANDOM", new DefaultPlaceRequest("DroolsDomainScreen"), authoringPerspectiveIdentifier).withSize(450).withLabel("Drools"),
+                new UberfireDock(UberfireDockPosition.EAST, "BRIEFCASE", new DefaultPlaceRequest("JPADomainScreen"), authoringPerspectiveIdentifier).withSize(450).withLabel("JPA"),
+                new UberfireDock(UberfireDockPosition.EAST, "COG", new DefaultPlaceRequest("AdvancedDomainScreen"), authoringPerspectiveIdentifier).withSize(450).withLabel("Advanced")
+
+        );
+        uberfireDocks.disable(UberfireDockPosition.EAST, authoringPerspectiveIdentifier);
+    }
+
+    public void onContextChange(@Observes DataModelerWorkbenchContextChangeEvent contextEvent) {
+        DataModelerContext context = dataModelerWBContext.getActiveContext();
+        if (shouldDisplayWestDocks(context)) {
+            uberfireDocks.enable(UberfireDockPosition.EAST, authoringPerspectiveIdentifier);
+        } else {
+            uberfireDocks.disable(UberfireDockPosition.EAST, authoringPerspectiveIdentifier);
+        }
+    }
+
+    private boolean shouldDisplayWestDocks(DataModelerContext context) {
+        return context != null && context.getEditionMode() == DataModelerContext.EditionMode.GRAPHICAL_MODE;
```

Comment:
@krisv as I told you that was my first time with Authoring Perspective, so I need help! :)

That rule comes from here https://github.com/droolsjbpm/kie-wb-common/blob/3e70fc6d8df3f9c589c54e6d2e52c8267d69b783/kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-client/src/main/java/org/kie/workbench/common/screens/datamodeller/client/widgets/common/domain/BaseDomainEditor.java#L123-L142

is that correct?